### PR TITLE
Chrome trace timestamp should be in microseconds not seconds.

### LINKED ISSUE #8599: Numba event mechanism is recording seconds as microseconds?
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

I'm using the Numba event mechanism to analyze a program and I notice the report generated by Numba is interpreting seconds as microseconds. So, to test this, I copied the [`ConstAddOne`](https://numba.pydata.org/numba-doc/latest/developer/custom_pipeline.html#compiler-pass-classes) from the docs and put a `time.sleep(6)` in the `run_pass` module. Below is the graph generated on [speedscope.app](speedscope.app) 

image

The code for the reproducer is in a [gist](https://gist.github.com/guilhermeleobas/a7263b571054c3d5c329d05b5aac277b)

### PR DESCRIPTION
**Pull request by sklam**
Fix #8599.

### CODE REVIEW

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. 

A potentially important side discussion is needed here as I don't think this is necessarily always a question of scaling microseconds to seconds. From the code, it looks like the timer used in the `event` module is based on `timeit.default_timer`. From the python docs, `timeit.default_timer` is always `time.perf_counter`, which doesn't seem to guarantee the unit of the returned value, it's just "the clock with the highest available resolution". This means the resolution needs to be known to work out an appropriate scaling factor to get to the unit of `seconds`.

I think there's two options:
1. Use `time.perf_counter_ns` for timing, time would then always be reported in nanoseconds and  scaling to seconds would always be via a constant factor. This also has the advantage of not losing precision through the use of a `float` type as the returned value.
2. Continue to use `timeit.default_timer` i.e. `time.perf_counter`. Then query `time.get_clock_info('perf_counter')` to get the resolution of the timer, the inverse of which is the scaling to get to the unit of `seconds`.

Any thoughts @sklam ?

**Comment by sklam** [comment]
@stuartarchibald, even though `timeit.default_timer` has unknown resolution but it is always in the unit of seconds. The new doc now say `defaut_timer` is always `time.perf_counter` which has `Return the value (in fractional seconds) of a performance counter`.

The tracing format expect a `float` type in microseconds unit anyway so i don't think the loss of precision matter that much. And, Numba is not that fast.

File: numba/core/event.py [hunk 1]
```diff
@@ -461,11 +461,13 @@ def _prepare_chrome_trace_data(listener: RecordingListener):
     for ts, rec in listener.buffer:
         data = rec.data
         cat = str(rec.kind)
+        ts_scaled = ts * 10 ** 6   # scale to microseconds
```
[the remark this answers is not in the document]
Comment (sklam) [reply numba/core/event.py:464]: oops

File: numba/core/event.py [hunk 2]
```diff
@@ -461,11 +461,13 @@ def _prepare_chrome_trace_data(listener: RecordingListener):
     for ts, rec in listener.buffer:
         data = rec.data
         cat = str(rec.kind)
+        ts_scaled = ts * 1_000_000   # scale to seconds
```
Comment (sklam) [inline numba/core/event.py:464]:
```suggestion
        ts_scaled = ts * 1_000_000   # scale to microseconds
```

File: numba/core/event.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sklam) [reply numba/core/event.py]: nvm, had to revert the change to the comments. not enough ☕

**Comment by stuartarchibald** [comment]
> @stuartarchibald, even though `timeit.default_timer` has unknown resolution but it is always in the unit of seconds. The new doc now say `defaut_timer` is always `time.perf_counter` which has `Return the value (in fractional seconds) of a performance counter`.
> 
> The tracing format expect a `float` type in microseconds unit anyway so i don't think the loss of precision matter that much. And, Numba is not that fast.

Thanks! Following some OOB discussion, turns out this is all ok. The issue is that the tracer output needs to be in microseconds and the clocks are reporting in seconds even if not explicitly noted, so the scale factor is constant and there's no resolution issue or otherwise!

**Review by stuartarchibald — APPROVED** [review]
Thanks for the updates, changes look good. @guilhermeleobas did you perhaps want to test this as in #8599 ?

**Comment by guilhermeleobas** [comment]
Thanks, I can confirm the fix solves the original issue.
image

**Comment by stuartarchibald** [comment]
Thanks for confirming @guilhermeleobas !

**Pull request merged**